todo_app/app.py

```diff
@@ -33,4 +60,25 @@ def item():
                 add_item(title)
         return redirect('/')
 
+    @app.route('/login/callback')
+    def login():
+        github_code = request.args.get('code')
+        params = {'client_id': config.github_app_id,
+                  'client_secret': config.github_app_secret,
+                  'code': github_code
+                  }
+        token_request = post('https://github.com/login/oauth/access_token',
+                             params=params, headers={'Accept': 'application/json'})
+        token = json.loads(token_request.text).get('access_token')
+        user_info_request = get('https://api.github.com/user', headers={
+                                'Authorization': f'Bearer {token}', 'Accept': 'application/json'})
+        user_info = json.loads(user_info_request.text)
+        user_id = user_info.get("id")
+        if user_id is None:
+            error('app did not authorise correctly')
+            return redirect('/error')
```

Comment:
prevents the app from repeatedly calling the auth service if something goes wrong which can cause rate limit issues